### PR TITLE
Update the PIP when it is not in the Succeeded provisioning state during the LB update.

### PR DESCRIPTION
**Pull request by nilo19**
**What type of PR is this?**

/kind bug

**What this PR does / why we need it**:
When the PIP is not in the `Succeeded` provisioning state, the LB update would fail. Here we perform an update to the PIP to ensure the provisioning state is `Succeeded`. In the next reconcile loop, the LB update would succeed.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
Update the PIP when it is not in the Succeeded provisioning state during the LB update.
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

File: staging/src/k8s.io/legacy-cloud-providers/azure/azure_backoff.go [hunk 1]
```diff
@@ -192,12 +199,40 @@ func (az *Cloud) CreateOrUpdateLB(service *v1.Service, lb network.LoadBalancer)
...
+		}
+		pipRG, pipName := matches[1], matches[2]
+		klog.V(3).Infof("The public IP %s referenced by load balancer %s is not in Succeeded provisioning state, will try to update it", pipName, to.String(lb.Name))
+		pip, _, err := az.getPublicIPAddress(pipRG, pipName)
+		if err != nil {
+			klog.Errorf("Failed to get the public IP %s in resource group %s: %v", pipName, pipRG, err)
+			return rerr.Error()
+		}
+		// Ensure the pip is in the Succeeded state and patch it
+		pip.ProvisioningState = to.StringPtr("Succeeded")
```
Comment (feiskyer) [inline staging/src/k8s.io/legacy-cloud-providers/azure/azure_backoff.go:228]: this line is not needed, please remove it

**Review by feiskyer — COMMENTED** [review]
/lgtm

**Comment by ialidzhikov** [comment]
@nilo19 , @feiskyer can we file cherry-picks of this PR?

**Comment by nilo19** [comment]
> @nilo19 , @feiskyer can we file cherry-picks of this PR?

sure, I'll do that.

**Pull request merged**